OpenRA.Mods.Common/Projectiles/AreaBeam.cs

```diff
@@ -132,8 +136,38 @@ public AreaBeam(AreaBeamInfo info, ProjectileArgs args, Color color)
 			length = Math.Max((target - headPos).Length / speed.Length, 1);
 		}
 
+		void UpdateTarget()
+		{
+			if (args.GuidedTarget.IsValidFor(args.SourceActor))
+			{
+				var guidedTargetPos = args.GuidedTarget.CenterPosition;
+				var targetDistance = (guidedTargetPos - args.Source).Length;
```

Comment:
You're comparing this against other `WDist`s, so you can/should avoid the `.Length` in these.